### ISSUE TITLE
Feature Request: LogSoftmax layer

### ISSUE THREAD

**Issue by fabiencro** [tensorflow/tensorflow#443]
Torch has a LogSoftmax layer, that does what the name imply: it is the equivalent of a softmax followed by a log. However, I could not find a similar layer in tensorflow.

LogSoftmax is quite convenient because I am often more interested in log-probabilities (eg for computing a log-likelihood) than in the probabilities themselves. 

Further, I suspect a LogSoftmax can be implemented more efficiently than a Softmax (it should at least saves a log call, if one is interested in the log-probabilities; plus it is less sensitive to underflow/overflows). Indeed Torch documentation indicates it is faster when one need the log-probabilities.

Would you consider adding such a layer at some point?

**Comment by girving** [comment]
Yep, this is a good feature request.

**Comment by kashif** [comment]
@girving can this issue be closed?

**Comment by girving** [comment]
Yep.  For future reference: If you include "Fixes #<issue>" in a commit message it'll autoclose.

**Comment by xudaaaaan** [comment]
Does the logsoftmax be added? Still I can't use logsoftmax
